dlt/extract/incremental/__init__.py

```diff
@@ -502,7 +502,13 @@ def __call__(self, rows: TDataItems, meta: Any = None) -> Optional[TDataItems]:
             # add directly computed hashes
             unique_hashes.update(transformer.unique_hashes)
             self._cached_state["unique_hashes"] = list(unique_hashes)
-
+            dedup_count = len(self._cached_state["unique_hashes"])
+            DEDUP_WARNING_THRESHOLD = 200
+            if dedup_count > DEDUP_WARNING_THRESHOLD:
+                logger.warning(
+                    f"There are {dedup_count} records to be deduplicated because"
+                    f" they share the same primary key `{self.primary_key}`."
```

Comment:
we need way better explanation here.
1. primary key is not known sometimes, then we hash the content. 
2. the root cause is not primary key. it is low resolution of cursor column. ie it is on a day. use cursor column name here

Reply:
Resolved in dd66d42